src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/mapView.js

```diff
@@ -68,29 +74,131 @@ class MapViewVisualization extends Visualization {
+        // TODO : Compute zoom somehow.
+        let zoomMatch      = parsedData.zoom !== undefined && parsedData.zoom !== null;
+        let zoom           = zoomMatch ? parsedData.zoom : computed.zoom;
 
         const deckgl = new deck.DeckGL({
             container: 'map',
-            mapboxApiAccessToken: 'pk.eyJ1IjoiZW5zby1vcmciLCJhIjoiY2tmNnh5MXh2MGlyOTJ5cWdubnFxbXo4ZSJ9.3KdAcCiiXJcSM18nwk09-Q',
+            mapboxApiAccessToken: TOKEN,
             mapStyle: parsedData.mapStyle || defaultMapStyle,
             initialViewState: {
-                longitude: parsedData.longitude || 0.0,
-                latitude: parsedData.latitude || 0.0,
-                zoom: parsedData.zoom || 3,
+                longitude: longitude,
+                latitude: latitude,
+                zoom: zoom,
                 pitch: parsedData.pitch || 0
             },
             controller: parsedData.controller || true
         });
 
         deckgl.setProps({
-            layers: parsedData.layers || []
+            layers: [scatterplotLayer]
+        });
+    }
+
+    /**
+     * Prepares data points to be shown on the map.
+     * @param preparedDataPoints - List holding geoPoints.
+     * @param parsedData - All the parsed data to create GeoPoints from.
+     * @param accentColor - accent color of IDE if element doesn't specify one.
+     */
+    prepareDataPoints(parsedData, preparedDataPoints, accentColor) {
+        let latitude  = 0.0;
+        let longitude = 0.0;
+        let zoom      = 11;
+
+        if (parsedData.type === GEO_POINT) {
+            this.pushGeoPoint(preparedDataPoints,parsedData,accentColor);
+            latitude  = parsedData.latitude;
+            longitude = parsedData.longitude;
+        } else if (Array.isArray(parsedData) && parsedData.length && parsedData[0].type === GEO_POINT) {
+            const computed = this.prepareDataPointsHelper(parsedData,preparedDataPoints,accentColor);
+            latitude       = computed.latitude;
+            longitude      = computed.longitude;
+        } else {
+            if (parsedData.type === SCATTERPLOT_LAYER && parsedData.data.length) {
+                const computed = this.prepareDataPointsHelper(parsedData.data,preparedDataPoints,accentColor);
+                latitude       = computed.latitude;
+                longitude      = computed.longitude;
+            } else if (parsedData.type === GEO_MAP && parsedData.layers !== undefined) {
+                parsedData.layers.forEach(layer => {
+                    if (layer.type === SCATTERPLOT_LAYER) {
+                        let dataPoints = layer.data || [];
+                        const computed = this.prepareDataPointsHelper(dataPoints,preparedDataPoints,accentColor);
+                        latitude       = computed.latitude;
+                        longitude      = computed.longitude;
+                    } else {
+                        console.log("Currently unsupported deck.gl layer.")
```

Comment:
prefix logs with name of the visualiztion - you can create `warn` function for it or something. In rust, we have done it automatically, here not yet.